enhancements/agent-installer/agent-based-installer.md

```diff
@@ -0,0 +1,328 @@
+with little guidance. The (hosted) Assisted service is not available for
+IPv6-only (until Quay.io supports IPv6) or disconnected environments. For some
+users, none of these options is optimal.
+
+On-premises users need a simple way to deploy their first cluster, even in a
+disconnected environment, using their own automation to boot hosts.
+
+### Goals
+
+- Install clusters in fully-disconnected environments.
+- Make use of assisted installation components to pre-validate the installation
+  environment.
+- Require no machines in the cluster environment other than those that are to
+  become part of the cluster.
+- Install clusters with single-node, compact, and high-availability topologies.
+- Be agnostic to the tools used to provision (virtual) machines so that users
+  can use their own (possibly pre-existing) tooling for provisioning.
+- Support install configurations with platform `none`, `baremetal`, or
+  `vsphere`.
+- Run on either Linux (any disro) or MacOS, in either x86-64 or arm64
+  architectures.
+- Install to machines with either x86-64 or arm64 architectures, independent of
+  the architecture the CLI tool runs on.
+
+### Non-Goals
+
+- Replace the IPI installer in any capacity.
+- Automate booting of the image on (virtual) machines.
+- Support install configurations for cloud-based platforms.
+- Provide an interactive API to drive automated deployments.
+- Operate in disconnected environments without the need to separately mirror
+  the release payload to a registry that is usable from the installed cluster.
+- Provide a mechanism to add and remove nodes from the cluster after it is
+  installed.
+- Generate image formats other than ISO.
+- Create minimal ISO images with a separate rootfs.
+- Run on Windows.
+
+## Proposal
+
+A command-line tool will enable users to build a custom RHCOS image, initially
+in ISO format, containing the components needed to install. The base ISO will
+be [retrieved](https://github.com/openshift/machine-os-images/) from a local
+container registry. Users can then boot this image on the hosts that are to
+become part of the cluster using any method they choose - this could be
+existing automation that they have or manually booting using physical media.
+One host (hereafter referred to as "Node 0") will be selected to run the
+assisted-service. All hosts will run the assisted agent. A single ISO will be
+used for all hosts in the cluster. Required container images will be pulled
+from the same local registry as the base ISO, as will the OpenShift release
+that is being installed.
+
+In the automated workflow, all configuration data for the cluster to be
+installed (including the expected number of hosts) will be provided as input to
+the command-line tool at the time the ISO is generated. Users will (optionally)
+be able to supply custom manifests to augment the initial configuration, in
+addition to data required for installation such as the network config.
```

Comment:
Will the input support the full range of what is possible with current Assisted Service CRs (ClusterDeployment, AgentClusterInstall, NMStateConfig, InfraEnv and extra installation manifests)?

Reply:
Great. Thanks!